packages/perennial/contracts/multiinvoker/MultiinvokerRollup.sol

```diff
@@ -0,0 +1,271 @@
+pragma solidity ^0.8.15;
+
+import "./MultiInvoker.sol";
+
+/// @title A calldata-optimized implementation of the Perennial MultiInvoker 
+/// @notice Retains same functionality and `invoke` entry point from inherited MultiInvoker
+contract MultiInvokerRollup is MultiInvoker {
+
+    event AddressAddedToCache(address indexed addr, uint256 nonce);
+
+    /// @dev reverts when address cache is relied on before a user / contract has been registered in state
+    /// if thrown, pass the bytes'00' (0 length) + the full 20 byte address instead of the length + index, 
+    /// can parse the events or mapping below to get the newly added cache index
+    error BadCalldata();
```

Comment:
For errors we use the format:

`ContractNameErrorNameError()`

So for this one it should be: `MultiInvokerRollupInvalidCalldataError()`.